src/msgpack/ArrayConverter.cs

```diff
@@ -94,7 +94,7 @@ private TArray ReadArray(IMsgPackReader reader, MsgPackContext context, uint len
             uint length,
             IMsgPackConverter<TElement> converter)
         {
-            var array = creator == null ? (TArray)Activator.CreateInstance(typeof(TArray)) : creator();
+            var array = creator == null ? context.GetObjectActivator<TArray>(typeof (TArray))() : creator();
```

Comment:
I was confused by code in TryGenerateMapConverter, but you are right, single generic parameter will be enough.